shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/args.py

```diff
@@ -1,52 +1,79 @@
 import os
-import urllib
+import re
+from urllib import parse
+from shipyard_templates import ShipyardLogger
 
+logger = ShipyardLogger.get_logger()
 
-def convert_to_boolean(string: str):
-    """
-    Shipyard can't support passing Booleans to code, so we have to convert
-    string values to their boolean values.
+
+def convert_to_boolean(string: str) -> bool:
     """
-    if string in ["True", "true", "TRUE"]:
-        value = True
-    else:
-        value = False
-    return value
+    Convert a string to a boolean.
 
+    Args:
+    string (str): A string to convert to a boolean.
 
-def set_environment_variables(environment_variables_to_set: str):
+    Returns:
+    bool: True if the string is "TRUE" (case-insensitive) else False.
     """
-    Set any args.value to an environment variable.
-    This will override system defaults.
-    Must be formatted as:
-    {'ENVIRONMENT_VARIABLE_NAME': args.value}
-    """
-    for environment_variable_name, arg_name in environment_variables_to_set.items():
-        os.environ[environment_variable_name] = arg_name
-    return
+    logger.debug(f"Converting {string} to boolean...")
+
+    clean = string.upper().strip()
+
+    if clean in {"TRUE", "FALSE"}:
+        result = clean == "TRUE"
+    else:
+        logger.error(f"String {string} is not a valid boolean.")
+        raise ValueError(
+            f"String {string} is not a valid boolean. Expected Rrue or False. Case-insensitive."
+        )
+
+    logger.debug(f"Converted {string} to {result} boolean.")
+    return result
 
 
-def create_shipyard_link():
+def create_shipyard_link() -> str:
```

Comment:
feels like the args folder wasn't the best place for this but not sure where a better home wouldve been. Open to suggestions. Only used in the email blueprints